design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,1110 @@
+
+### Life Cycle and Updates
+
+#### Resource life cycle
+
+On each reconciliation loop for a TSM managed kind, the rendering process of the
+TSM will use a single pass at applying templates.
+
+* Get the current resource
+* Get the `templates` from the `Stack` resource that owns this CRD
+* All of the dependent resources (generated from templates) are fetched.
+* The current resource and dependent resources are combined to form variables to
+  pass to the template.
+* Template substitutions are made for each resource and the `templateStatus`
+* All of the dependent, template created, resources are applied to the API,
+  including a pass at deleting resources with empty template results.
+
+Because this process operates in a single read then write pass, it can not get
+caught in a circular dependency loop. However, Template Stack creators should
+avoid crafting their templates in a way that they would never resolve as
+intended.
+
+The rendered template will replace changes to dependent resources that are found
+in the Kubernetes API. More importantly, since the values are obtained directly
+from the API on each pass, if a resource is updated by a user or some other
+process, those values will be used when rendering.
+
+If a templates rendering contains invalid resource YAML, or contains fields and
+values are not valid for the resource being generated, the Template Stack
+Manager will need a defined means of surfacing these error conditions. Just as
+sibling resources should be made available through templates, the error messages
+should be made available through templates. The implementation of the TSM will
+determine if this is done through functions or structured values. This approach
+lets Template Stack and CRD authors embed status conditions in a way that best
+meets the needs of their application.
+
+Error conditions will also be logged by the TSM and pushed to the Kubernetes API
+as Events.
+
+*The crossplane-cli could be adapted to detect these conditions but that is not
+a concern that will be discussed in this design.*
+
+##### Watches and Timers
+
+The TSM is responsible for updating the `Foo` (Stack resource) and
+`DependentThing` template resources. Let's explore how the TSM is triggered.
+
+The TSM will
+[watch](https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes)
+the resources belonging to its set of CRDs (`Foo`). But it does not need to
+detect changes on the dependent resources, in this same way.
+
+Rather, a
+[`Result{RequeueAfter}`](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/reconcile/reconcile.go#L30-L32)
+on `Foo` would be sufficient for an initial implementation of Template Stacks.
+
+On each reconciliation, the dependent resources will be fetched (for use as
+template variables), and then the template will be rendered, and the dependent
+resources will be updated (or created or deleted).
+
+There should be advantages to [watching each dependent
```

Comment:
we'll definitely want to improve on a timer based approach not too long after initial implementation both for improved user experience and decreased API server load.